{
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "view-in-github",
        "colab_type": "text"
      },
      "source": [
        "<a href=\"https://colab.research.google.com/github/bhavya-kashyap/AMC_crosswalk/blob/main/abstract_pubmed_webscrapper.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!apt update\n",
        "!apt install chromium-chromedriver\n",
        "!pip install selenium\n",
        "!pip install bs4\n",
        "!pip install pymed"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "XoBK_wpnJQJB",
        "outputId": "64edcf4e-abd1-4a43-e414-a8f833bcd51c"
      },
      "id": "XoBK_wpnJQJB",
      "execution_count": 1,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\u001b[33m\r0% [Working]\u001b[0m\r            \rGet:1 http://security.ubuntu.com/ubuntu bionic-security InRelease [88.7 kB]\n",
            "\u001b[33m\r0% [Waiting for headers] [1 InRelease 14.2 kB/88.7 kB 16%] [Connected to cloud.\u001b[0m\r                                                                               \rHit:2 http://archive.ubuntu.com/ubuntu bionic InRelease\n",
            "\u001b[33m\r0% [Waiting for headers] [1 InRelease 34.4 kB/88.7 kB 39%] [Connected to cloud.\u001b[0m\u001b[33m\r0% [2 InRelease gpgv 242 kB] [Waiting for headers] [1 InRelease 37.3 kB/88.7 kB\u001b[0m\r                                                                               \rGet:3 https://cloud.r-project.org/bin/linux/ubuntu bionic-cran40/ InRelease [3,626 B]\n",
            "Get:4 http://archive.ubuntu.com/ubuntu bionic-updates InRelease [88.7 kB]\n",
            "Ign:5 https://developer.download.nvidia.com/compute/machine-learning/repos/ubuntu1804/x86_64  InRelease\n",
            "Get:6 http://ppa.launchpad.net/c2d4u.team/c2d4u4.0+/ubuntu bionic InRelease [15.9 kB]\n",
            "Hit:7 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64  InRelease\n",
            "Hit:8 https://developer.download.nvidia.com/compute/machine-learning/repos/ubuntu1804/x86_64  Release\n",
            "Get:9 http://archive.ubuntu.com/ubuntu bionic-backports InRelease [83.3 kB]\n",
            "Hit:10 http://ppa.launchpad.net/cran/libgit2/ubuntu bionic InRelease\n",
            "Get:11 http://ppa.launchpad.net/deadsnakes/ppa/ubuntu bionic InRelease [15.9 kB]\n",
            "Hit:12 http://ppa.launchpad.net/graphics-drivers/ppa/ubuntu bionic InRelease\n",
            "Get:13 http://security.ubuntu.com/ubuntu bionic-security/restricted amd64 Packages [1,217 kB]\n",
            "Get:14 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 Packages [3,467 kB]\n",
            "Get:15 http://archive.ubuntu.com/ubuntu bionic-updates/universe amd64 Packages [2,332 kB]\n",
            "Get:16 http://archive.ubuntu.com/ubuntu bionic-updates/restricted amd64 Packages [1,257 kB]\n",
            "Get:18 http://ppa.launchpad.net/c2d4u.team/c2d4u4.0+/ubuntu bionic/main Sources [2,169 kB]\n",
            "Get:19 http://ppa.launchpad.net/c2d4u.team/c2d4u4.0+/ubuntu bionic/main amd64 Packages [1,111 kB]\n",
            "Fetched 11.8 MB in 4s (2,814 kB/s)\n",
            "Reading package lists... Done\n",
            "Building dependency tree       \n",
            "Reading state information... Done\n",
            "36 packages can be upgraded. Run 'apt list --upgradable' to see them.\n",
            "Reading package lists... Done\n",
            "Building dependency tree       \n",
            "Reading state information... Done\n",
            "The following package was automatically installed and is no longer required:\n",
            "  libnvidia-common-460\n",
            "Use 'apt autoremove' to remove it.\n",
            "The following additional packages will be installed:\n",
            "  chromium-browser chromium-browser-l10n chromium-codecs-ffmpeg-extra\n",
            "Suggested packages:\n",
            "  webaccounts-chromium-extension unity-chromium-extension\n",
            "The following NEW packages will be installed:\n",
            "  chromium-browser chromium-browser-l10n chromium-chromedriver\n",
            "  chromium-codecs-ffmpeg-extra\n",
            "0 upgraded, 4 newly installed, 0 to remove and 36 not upgraded.\n",
            "Need to get 91.7 MB of archives.\n",
            "After this operation, 309 MB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu bionic-updates/universe amd64 chromium-codecs-ffmpeg-extra amd64 105.0.5195.102-0ubuntu0.18.04.1 [1,156 kB]\n",
            "Get:2 http://archive.ubuntu.com/ubuntu bionic-updates/universe amd64 chromium-browser amd64 105.0.5195.102-0ubuntu0.18.04.1 [80.1 MB]\n",
            "Get:3 http://archive.ubuntu.com/ubuntu bionic-updates/universe amd64 chromium-browser-l10n all 105.0.5195.102-0ubuntu0.18.04.1 [5,097 kB]\n",
            "Get:4 http://archive.ubuntu.com/ubuntu bionic-updates/universe amd64 chromium-chromedriver amd64 105.0.5195.102-0ubuntu0.18.04.1 [5,320 kB]\n",
            "Fetched 91.7 MB in 2s (47.2 MB/s)\n",
            "Selecting previously unselected package chromium-codecs-ffmpeg-extra.\n",
            "(Reading database ... 123942 files and directories currently installed.)\n",
            "Preparing to unpack .../chromium-codecs-ffmpeg-extra_105.0.5195.102-0ubuntu0.18.04.1_amd64.deb ...\n",
            "Unpacking chromium-codecs-ffmpeg-extra (105.0.5195.102-0ubuntu0.18.04.1) ...\n",
            "Selecting previously unselected package chromium-browser.\n",
            "Preparing to unpack .../chromium-browser_105.0.5195.102-0ubuntu0.18.04.1_amd64.deb ...\n",
            "Unpacking chromium-browser (105.0.5195.102-0ubuntu0.18.04.1) ...\n",
            "Selecting previously unselected package chromium-browser-l10n.\n",
            "Preparing to unpack .../chromium-browser-l10n_105.0.5195.102-0ubuntu0.18.04.1_all.deb ...\n",
            "Unpacking chromium-browser-l10n (105.0.5195.102-0ubuntu0.18.04.1) ...\n",
            "Selecting previously unselected package chromium-chromedriver.\n",
            "Preparing to unpack .../chromium-chromedriver_105.0.5195.102-0ubuntu0.18.04.1_amd64.deb ...\n",
            "Unpacking chromium-chromedriver (105.0.5195.102-0ubuntu0.18.04.1) ...\n",
            "Setting up chromium-codecs-ffmpeg-extra (105.0.5195.102-0ubuntu0.18.04.1) ...\n",
            "Setting up chromium-browser (105.0.5195.102-0ubuntu0.18.04.1) ...\n",
            "update-alternatives: using /usr/bin/chromium-browser to provide /usr/bin/x-www-browser (x-www-browser) in auto mode\n",
            "update-alternatives: using /usr/bin/chromium-browser to provide /usr/bin/gnome-www-browser (gnome-www-browser) in auto mode\n",
            "Setting up chromium-chromedriver (105.0.5195.102-0ubuntu0.18.04.1) ...\n",
            "Setting up chromium-browser-l10n (105.0.5195.102-0ubuntu0.18.04.1) ...\n",
            "Processing triggers for man-db (2.8.3-2ubuntu0.1) ...\n",
            "Processing triggers for hicolor-icon-theme (0.17-2) ...\n",
            "Processing triggers for mime-support (3.60ubuntu1) ...\n",
            "Processing triggers for libc-bin (2.27-3ubuntu1.6) ...\n",
            "Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/\n",
            "Collecting selenium\n",
            "  Downloading selenium-4.5.0-py3-none-any.whl (995 kB)\n",
            "\u001b[K     |████████████████████████████████| 995 kB 5.1 MB/s \n",
            "\u001b[?25hCollecting urllib3[socks]~=1.26\n",
            "  Downloading urllib3-1.26.12-py2.py3-none-any.whl (140 kB)\n",
            "\u001b[K     |████████████████████████████████| 140 kB 53.0 MB/s \n",
            "\u001b[?25hRequirement already satisfied: certifi>=2021.10.8 in /usr/local/lib/python3.7/dist-packages (from selenium) (2022.9.24)\n",
            "Collecting trio-websocket~=0.9\n",
            "  Downloading trio_websocket-0.9.2-py3-none-any.whl (16 kB)\n",
            "Collecting trio~=0.17\n",
            "  Downloading trio-0.22.0-py3-none-any.whl (384 kB)\n",
            "\u001b[K     |████████████████████████████████| 384 kB 42.6 MB/s \n",
            "\u001b[?25hCollecting async-generator>=1.9\n",
            "  Downloading async_generator-1.10-py3-none-any.whl (18 kB)\n",
            "Requirement already satisfied: sortedcontainers in /usr/local/lib/python3.7/dist-packages (from trio~=0.17->selenium) (2.4.0)\n",
            "Requirement already satisfied: idna in /usr/local/lib/python3.7/dist-packages (from trio~=0.17->selenium) (2.10)\n",
            "Collecting sniffio\n",
            "  Downloading sniffio-1.3.0-py3-none-any.whl (10 kB)\n",
            "Requirement already satisfied: attrs>=19.2.0 in /usr/local/lib/python3.7/dist-packages (from trio~=0.17->selenium) (22.1.0)\n",
            "Collecting exceptiongroup>=1.0.0rc9\n",
            "  Downloading exceptiongroup-1.0.0-py3-none-any.whl (12 kB)\n",
            "Collecting outcome\n",
            "  Downloading outcome-1.2.0-py2.py3-none-any.whl (9.7 kB)\n",
            "Collecting wsproto>=0.14\n",
            "  Downloading wsproto-1.2.0-py3-none-any.whl (24 kB)\n",
            "Requirement already satisfied: PySocks!=1.5.7,<2.0,>=1.5.6 in /usr/local/lib/python3.7/dist-packages (from urllib3[socks]~=1.26->selenium) (1.7.1)\n",
            "Collecting h11<1,>=0.9.0\n",
            "  Downloading h11-0.14.0-py3-none-any.whl (58 kB)\n",
            "\u001b[K     |████████████████████████████████| 58 kB 6.4 MB/s \n",
            "\u001b[?25hRequirement already satisfied: typing-extensions in /usr/local/lib/python3.7/dist-packages (from h11<1,>=0.9.0->wsproto>=0.14->trio-websocket~=0.9->selenium) (4.1.1)\n",
            "Installing collected packages: sniffio, outcome, h11, exceptiongroup, async-generator, wsproto, urllib3, trio, trio-websocket, selenium\n",
            "  Attempting uninstall: urllib3\n",
            "    Found existing installation: urllib3 1.24.3\n",
            "    Uninstalling urllib3-1.24.3:\n",
            "      Successfully uninstalled urllib3-1.24.3\n",
            "\u001b[31mERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\n",
            "requests 2.23.0 requires urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1, but you have urllib3 1.26.12 which is incompatible.\u001b[0m\n",
            "Successfully installed async-generator-1.10 exceptiongroup-1.0.0 h11-0.14.0 outcome-1.2.0 selenium-4.5.0 sniffio-1.3.0 trio-0.22.0 trio-websocket-0.9.2 urllib3-1.26.12 wsproto-1.2.0\n",
            "Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/\n",
            "Requirement already satisfied: bs4 in /usr/local/lib/python3.7/dist-packages (0.0.1)\n",
            "Requirement already satisfied: beautifulsoup4 in /usr/local/lib/python3.7/dist-packages (from bs4) (4.6.3)\n",
            "Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/\n",
            "Collecting pymed\n",
            "  Downloading pymed-0.8.9-py3-none-any.whl (9.6 kB)\n",
            "Requirement already satisfied: requests>=2.20.0 in /usr/local/lib/python3.7/dist-packages (from pymed) (2.23.0)\n",
            "Requirement already satisfied: idna<3,>=2.5 in /usr/local/lib/python3.7/dist-packages (from requests>=2.20.0->pymed) (2.10)\n",
            "Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.7/dist-packages (from requests>=2.20.0->pymed) (2022.9.24)\n",
            "Collecting urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1\n",
            "  Downloading urllib3-1.25.11-py2.py3-none-any.whl (127 kB)\n",
            "\u001b[K     |████████████████████████████████| 127 kB 4.8 MB/s \n",
            "\u001b[?25hRequirement already satisfied: chardet<4,>=3.0.2 in /usr/local/lib/python3.7/dist-packages (from requests>=2.20.0->pymed) (3.0.4)\n",
            "Installing collected packages: urllib3, pymed\n",
            "  Attempting uninstall: urllib3\n",
            "    Found existing installation: urllib3 1.26.12\n",
            "    Uninstalling urllib3-1.26.12:\n",
            "      Successfully uninstalled urllib3-1.26.12\n",
            "\u001b[31mERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\n",
            "selenium 4.5.0 requires urllib3[socks]~=1.26, but you have urllib3 1.25.11 which is incompatible.\u001b[0m\n",
            "Successfully installed pymed-0.8.9 urllib3-1.25.11\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 2,
      "id": "11409abe",
      "metadata": {
        "id": "11409abe"
      },
      "outputs": [],
      "source": [
        "import pandas as pd\n",
        "import numpy as np\n",
        "import pymed\n",
        "from selenium import webdriver\n",
        "from selenium.webdriver.chrome.options import Options\n",
        "from selenium.webdriver.common.by import By\n",
        "from bs4 import BeautifulSoup\n",
        "import time\n",
        "import math"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 3,
      "id": "0f4e54fd",
      "metadata": {
        "id": "0f4e54fd"
      },
      "outputs": [],
      "source": [
        "# Data parameters\n",
        "# https://www.webofscience.com/wos/woscc/summary/f05a3dd0-ac52-4364-859d-aea68cf4f4d3-583fd5ef/relevance/1\n",
        "# SO = (MMWR RECOMMENDATIONS AND REPORTS OR NEW ENGLAND JOURNAL OF MEDICINE OR NATURE MEDICINE OR NATURE REVIEWS IMMUNOLOGY OR MORBIDITY AND MORTALITY WEEKLY REPORT OR LANCET OR PHYSIOLOGICAL REVIEWS OR NATURE REVIEWS DRUG DISCOVERY OR ANNUAL REVIEW OF PATHOLOGY: MECHANISMS OF DISEASE OR JOURNAL OF CLINICAL ONCOLOGY OR LANCET GLOBAL HEALTH OR ANNALS OF ONCOLOGY OR ANNUAL REVIEW OF PUBLIC HEALTH OR NATURE REVIEWS DISEASE PRIMERS OR ACCOUNTS OF CHEMICAL RESEARCH OR ANNUAL REVIEW OF CLINICAL PSYCHOLOGY OR JOURNAL OF EXPERIMENTAL MEDICINE OR MOLECULAR SYSTEMS BIOLOGY OR ANNUAL REVIEW OF MICROBIOLOGY OR JAMA - JOURNAL OF THE AMERICAN MEDICAL ASSOCIATION OR LANCET DIGITAL HEALTH OR REPORTS ON PROGRESS IN PHYSICS OR NATURE BIOMEDICAL ENGINEERING OR EMBO JOURNAL OR JAMA PSYCHIATRY OR NATURE REVIEWS MOLECULAR CELL BIOLOGY OR CELL OR NATURE REVIEWS GENETICS OR NATURE BIOTECHNOLOGY OR NATURE REVIEWS CANCER OR NATURE GENETICS OR ANNUAL REVIEW OF BIOCHEMISTRY OR NATURE CATALYSIS OR NATURE METHODS OR CANCER CELL OR MOLECULAR CELL OR NATURE CELL BIOLOGY OR LANCET DIABETES AND ENDOCRINOLOGY OR NATURE MICROBIOLOGY OR ANNUAL REVIEW OF PLANT BIOLOGY OR ANNUAL REVIEW OF GENETICS OR CELL HOST AND MICROBE OR GENOME BIOLOGY OR CELL METABOLISM OR NATURE REVIEWS ENDOCRINOLOGY OR NATURE STRUCTURAL AND MOLECULAR BIOLOGY OR NUCLEIC ACIDS RESEARCH OR PHARMACOLOGICAL REVIEWS OR ANNUAL REVIEW OF PHARMACOLOGY AND TOXICOLOGY OR NANO TODAY OR DRUG RESISTANCE UPDATES OR JOURNAL FOR IMMUNOTHERAPY OF CANCER OR TRENDS IN PHARMACOLOGICAL SCIENCES OR EMERGING MICROBES AND INFECTIONS OR PHARMACOLOGY AND THERAPEUTICS OR MOLECULAR THERAPY OR NPJ VACCINES OR ADVANCED DRUG DELIVERY REVIEWS OR NATURAL PRODUCT REPORTS OR CELL CHEMICAL BIOLOGY OR PROTEIN AND CELL OR CELLULAR AND MOLECULAR LIFE SCIENCES OR MEDICINAL RESEARCH REVIEWS OR NEUROPSYCHOPHARMACOLOGY OR ANTIVIRAL RESEARCH OR ACTA PHARMACEUTICA SINICA B OR PAIN OR THERANOSTICS OR ADVANCED HEALTHCARE MATERIALS OR BRITISH JOURNAL OF PHARMACOLOGY OR ENVIRONMENTAL POLLUTION OR ALZHEIMER'S AND DEMENTIA OR HEALTH AFFAIRS OR LANCET PLANETARY HEALTH OR IMPLEMENTATION SCIENCE OR INFECTIOUS DISEASE MODELLING OR BMJ GLOBAL HEALTH OR GLOBALIZATION AND HEALTH OR JOURNAL OF HEALTH ECONOMICS OR JOURNAL OF THE AMERICAN MEDICAL DIRECTORS ASSOCIATION OR MILBANK QUARTERLY OR EUROPEAN HEART JOURNAL. QUALITY OF CARE &AMP, CLINICAL OUTCOMES OR BMJ QUALITY AND SAFETY OR HEALTH SERVICES RESEARCH OR MEDICAL CARE RESEARCH AND REVIEW OR VALUE IN HEALTH OR POLICY AND INTERNET OR INTERNATIONAL JOURNAL OF DRUG POLICY OR INTERNATIONAL JOURNAL FOR EQUITY IN HEALTH OR JOURNAL OF GLOBAL HEALTH OR PHARMACOECONOMICS OR HEALTH RESEARCH POLICY AND SYSTEMS OR GLOBAL HEALTH RESEARCH AND POLICY OR INTERNATIONAL JOURNAL OF TRANSGENDER HEALTH OR JCO ONCOLOGY PRACTICE OR PREVENTING CHRONIC DISEASE OR NATURE OR SCIENCE OR SCIENCE ADVANCES OR PROCEEDINGS OF THE NATIONAL ACADEMY OF SCIENCES OF THE UNITED STATES OF AMERICA OR NATIONAL SCIENCE REVIEW OR SCIENCE BULLETIN OR RESEARCH OR JOURNAL OF ADVANCED RESEARCH OR ISCIENCE OR SCIENTIFIC REPORTS OR TSINGHUA SCIENCE AND TECHNOLOGY OR PROCEEDINGS OF THE JAPAN ACADEMY SERIES B: PHYSICAL AND BIOLOGICAL SCIENCES OR ADVANCED THEORY AND SIMULATIONS OR PROGRESS IN NATURAL SCIENCE: MATERIALS INTERNATIONAL OR PLOS ONE OR SCIENCE AND TECHNOLOGY STUDIES OR ROYAL SOCIETY OPEN SCIENCE OR RESEARCH IN SCIENCE AND TECHNOLOGICAL EDUCATION OR INTERNATIONAL JOURNAL OF BIFURCATION AND CHAOS IN APPLIED SCIENCES AND ENGINEERING OR LATIN AMERICAN RESEARCH REVIEW OR FRACTALS OR FACETS OR HELIYON OR JOURNAL OF KING SAUD UNIVERSITY - SCIENCE OR ARABIAN JOURNAL FOR SCIENCE AND ENGINEERING OR CLINICAL AND TRANSLATIONAL IMMUNOLOGY OR JOURNAL OF PAIN AND SYMPTOM MANAGEMENT OR INTERNATIONAL JOURNAL OF NURSING STUDIES OR NURSE EDUCATION TODAY OR JOURNAL OF NURSING SCHOLARSHIP OR JOURNAL OF PALLIATIVE MEDICINE OR PATIENT OR CLINICAL SIMULATION IN NURSING OR INTERNATIONAL NURSING REVIEW OR JOURNAL OF CLINICAL NURSING OR WORLDVIEWS ON EVIDENCE-BASED NURSING OR NURSE EDUCATION IN PRACTICE OR JOURNAL OF ADVANCED NURSING OR JOURNAL OF SCHOOL NURSING OR JOURNAL OF RELIGION AND HEALTH OR INTERNATIONAL JOURNAL OF NURSING SCIENCES OR NURSING OUTLOOK OR PALLIATIVE AND SUPPORTIVE CARE OR SYMBOLIC INTERACTION OR JOURNAL OF PROFESSIONAL NURSING OR JOURNAL OF TRANSCULTURAL NURSING OR JOURNAL OF NURSING EDUCATION OR BMC NURSING OR JOURNAL OF MULTIDISCIPLINARY HEALTHCARE)\n",
        "# Publication years: last 5 years (2018 - 2022)\n",
        "# Affiliations: Emory university\n",
        "# Size of data: 1000 records\n",
        "# primary focus is to get the pubmed ID"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 7,
      "id": "4b429684",
      "metadata": {
        "id": "4b429684"
      },
      "outputs": [],
      "source": [
        "df = pd.read_csv(\"sampledata.csv\")\n",
        "df.shape"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 9,
      "id": "e86a01cd",
      "metadata": {
        "id": "e86a01cd"
      },
      "outputs": [],
      "source": [
        "# df.head()\n",
        "df.drop(columns = ['Publication Type','Authors','Author Full Names', 'Group Authors','Source Title','Document Type', 'Publication Date','Publication Year', 'Issue','Start Page','End Page', 'Article Number','DOI','Volume', 'DOI Link', 'Early Access Date', 'UT (Unique WOS ID)', 'Web of Science Record'], inplace=True)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 10,
      "id": "fe199406",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 363
        },
        "id": "fe199406",
        "outputId": "1a109505-dccc-41c0-f217-8cfbb514a9f7"
      },
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "                                       Article Title  \\\n",
              "0  A low-cost genomics workflow enables isolate s...   \n",
              "1  A saturated map of common genetic variants ass...   \n",
              "2  Evaluation of a non-pharmacological interdisci...   \n",
              "3  Comparative genomic and biochemical analyses i...   \n",
              "4  Africa must participate in finding a gene ther...   \n",
              "5  ADHESION G PROTEIN-COUPLED RECEPTORS: STRUCTUR...   \n",
              "6  Authorship equity guidelines in global health ...   \n",
              "7  Stroke genetics informs drug discovery and ris...   \n",
              "8  Toward a Uniform Classification of Nurse Pract...   \n",
              "9  p38MAPK guards the integrity of endosomal comp...   \n",
              "\n",
              "                                            Abstract   Pubmed Id  \\\n",
              "0  Earth's environments harbor complex consortia ...  36224660.0   \n",
              "1  Common single-nucleotide polymorphisms (SNPs) ...  36224396.0   \n",
              "2  Objective To conduct a quality improvement eva...  36064854.0   \n",
              "3  Humans and Acanthamoeba polyphaga mimivirus sh...  36207453.0   \n",
              "4                                                NaN  36203001.0   \n",
              "5  Adhesion G protein-coupled receptors (AGPCRs) ...  35468004.0   \n",
              "6                                                NaN  36220309.0   \n",
              "7  Previous genome-wide association studies (GWAS...  36180795.0   \n",
              "8  Many states' scope of practice laws limits the...  36172783.0   \n",
              "9  Pathogens trigger activation of sensors of the...  36175595.0   \n",
              "\n",
              "                                         URL  \n",
              "0  https://pubmed.ncbi.nlm.nih.gov/36224660/  \n",
              "1  https://pubmed.ncbi.nlm.nih.gov/36224396/  \n",
              "2  https://pubmed.ncbi.nlm.nih.gov/36064854/  \n",
              "3  https://pubmed.ncbi.nlm.nih.gov/36207453/  \n",
              "4  https://pubmed.ncbi.nlm.nih.gov/36203001/  \n",
              "5  https://pubmed.ncbi.nlm.nih.gov/35468004/  \n",
              "6  https://pubmed.ncbi.nlm.nih.gov/36220309/  \n",
              "7  https://pubmed.ncbi.nlm.nih.gov/36180795/  \n",
              "8  https://pubmed.ncbi.nlm.nih.gov/36172783/  \n",
              "9  https://pubmed.ncbi.nlm.nih.gov/36175595/  "
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-74656131-c97e-4a4d-837e-353b3fcdda9a\">\n",
              "    <div class=\"colab-df-container\">\n",
              "      <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>Article Title</th>\n",
              "      <th>Abstract</th>\n",
              "      <th>Pubmed Id</th>\n",
              "      <th>URL</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>A low-cost genomics workflow enables isolate s...</td>\n",
              "      <td>Earth's environments harbor complex consortia ...</td>\n",
              "      <td>36224660.0</td>\n",
              "      <td>https://pubmed.ncbi.nlm.nih.gov/36224660/</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>A saturated map of common genetic variants ass...</td>\n",
              "      <td>Common single-nucleotide polymorphisms (SNPs) ...</td>\n",
              "      <td>36224396.0</td>\n",
              "      <td>https://pubmed.ncbi.nlm.nih.gov/36224396/</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>Evaluation of a non-pharmacological interdisci...</td>\n",
              "      <td>Objective To conduct a quality improvement eva...</td>\n",
              "      <td>36064854.0</td>\n",
              "      <td>https://pubmed.ncbi.nlm.nih.gov/36064854/</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>Comparative genomic and biochemical analyses i...</td>\n",
              "      <td>Humans and Acanthamoeba polyphaga mimivirus sh...</td>\n",
              "      <td>36207453.0</td>\n",
              "      <td>https://pubmed.ncbi.nlm.nih.gov/36207453/</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>Africa must participate in finding a gene ther...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>36203001.0</td>\n",
              "      <td>https://pubmed.ncbi.nlm.nih.gov/36203001/</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5</th>\n",
              "      <td>ADHESION G PROTEIN-COUPLED RECEPTORS: STRUCTUR...</td>\n",
              "      <td>Adhesion G protein-coupled receptors (AGPCRs) ...</td>\n",
              "      <td>35468004.0</td>\n",
              "      <td>https://pubmed.ncbi.nlm.nih.gov/35468004/</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>6</th>\n",
              "      <td>Authorship equity guidelines in global health ...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>36220309.0</td>\n",
              "      <td>https://pubmed.ncbi.nlm.nih.gov/36220309/</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>7</th>\n",
              "      <td>Stroke genetics informs drug discovery and ris...</td>\n",
              "      <td>Previous genome-wide association studies (GWAS...</td>\n",
              "      <td>36180795.0</td>\n",
              "      <td>https://pubmed.ncbi.nlm.nih.gov/36180795/</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>8</th>\n",
              "      <td>Toward a Uniform Classification of Nurse Pract...</td>\n",
              "      <td>Many states' scope of practice laws limits the...</td>\n",
              "      <td>36172783.0</td>\n",
              "      <td>https://pubmed.ncbi.nlm.nih.gov/36172783/</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>9</th>\n",
              "      <td>p38MAPK guards the integrity of endosomal comp...</td>\n",
              "      <td>Pathogens trigger activation of sensors of the...</td>\n",
              "      <td>36175595.0</td>\n",
              "      <td>https://pubmed.ncbi.nlm.nih.gov/36175595/</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "      <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-74656131-c97e-4a4d-837e-353b3fcdda9a')\"\n",
              "              title=\"Convert this dataframe to an interactive table.\"\n",
              "              style=\"display:none;\">\n",
              "        \n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M0 0h24v24H0V0z\" fill=\"none\"/>\n",
              "    <path d=\"M18.56 5.44l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94zm-11 1L8.5 8.5l.94-2.06 2.06-.94-2.06-.94L8.5 2.5l-.94 2.06-2.06.94zm10 10l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94z\"/><path d=\"M17.41 7.96l-1.37-1.37c-.4-.4-.92-.59-1.43-.59-.52 0-1.04.2-1.43.59L10.3 9.45l-7.72 7.72c-.78.78-.78 2.05 0 2.83L4 21.41c.39.39.9.59 1.41.59.51 0 1.02-.2 1.41-.59l7.78-7.78 2.81-2.81c.8-.78.8-2.07 0-2.86zM5.41 20L4 18.59l7.72-7.72 1.47 1.35L5.41 20z\"/>\n",
              "  </svg>\n",
              "      </button>\n",
              "      \n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      flex-wrap:wrap;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "      <script>\n",
              "        const buttonEl =\n",
              "          document.querySelector('#df-74656131-c97e-4a4d-837e-353b3fcdda9a button.colab-df-convert');\n",
              "        buttonEl.style.display =\n",
              "          google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "        async function convertToInteractive(key) {\n",
              "          const element = document.querySelector('#df-74656131-c97e-4a4d-837e-353b3fcdda9a');\n",
              "          const dataTable =\n",
              "            await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                     [key], {});\n",
              "          if (!dataTable) return;\n",
              "\n",
              "          const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "            '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "            + ' to learn more about interactive tables.';\n",
              "          element.innerHTML = '';\n",
              "          dataTable['output_type'] = 'display_data';\n",
              "          await google.colab.output.renderOutput(dataTable, element);\n",
              "          const docLink = document.createElement('div');\n",
              "          docLink.innerHTML = docLinkHtml;\n",
              "          element.appendChild(docLink);\n",
              "        }\n",
              "      </script>\n",
              "    </div>\n",
              "  </div>\n",
              "  "
            ]
          },
          "metadata": {},
          "execution_count": 10
        }
      ],
      "source": [
        "baseurl = 'https://pubmed.ncbi.nlm.nih.gov/'\n",
        "urls = []\n",
        "for i in range(len(df)):\n",
        "    urls.append(baseurl + str(df.loc[i]['Pubmed Id'].astype(int)) + '/')\n",
        "df['URL'] = urls\n",
        "df.head(10)"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "sample = df.head(20)"
      ],
      "metadata": {
        "id": "wzONJeCfhhnr"
      },
      "id": "wzONJeCfhhnr",
      "execution_count": 11,
      "outputs": []
    },
    {
      "cell_type": "code",
      "execution_count": 12,
      "id": "eee408f2",
      "metadata": {
        "id": "eee408f2"
      },
      "outputs": [],
      "source": [
        "def driversetup():\n",
        "      options = webdriver.ChromeOptions()\n",
        "      #run Selenium in headless mode\n",
        "      options.add_argument('--headless')\n",
        "      options.add_argument('--no-sandbox')\n",
        "      options.add_argument(\"enable-automation\")\n",
        "      #overcome limited resource problems\n",
        "      options.add_argument('--disable-dev-shm-usage')\n",
        "      options.add_argument(\"lang=en\")\n",
        "      #open Browser in maximized mode\n",
        "      options.add_argument(\"start-maximized\")\n",
        "      #disable infobars\n",
        "      options.add_argument(\"disable-infobars\")\n",
        "      #disable extension\n",
        "      options.add_argument(\"--disable-browser-side-navigation\")\n",
        "      options.add_argument(\"--disable-gpu\")\n",
        "      options.add_argument(\"--disable-extensions\")\n",
        "      options.add_argument(\"--incognito\")\n",
        "      options.add_argument(\"--disable-blink-features=AutomationControlled\")\n",
        "      driver = webdriver.Chrome(options=options)\n",
        "      driver.execute_script(\"Object.defineProperty(navigator, 'webdriver', {get: () => undefined});\")\n",
        "      return driver\n",
        "\n",
        "def pagesource(url, driver):\n",
        "      driver = driver\n",
        "      driver.get(url)\n",
        "      time.sleep(8)\n",
        "      soup = BeautifulSoup(driver.page_source)\n",
        "      return soup"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 14,
      "id": "2d7f99e2",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "2d7f99e2",
        "outputId": "44f9fe92-7b19-45b2-f502-c316c323ec5a"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "0  \n",
            "\n",
            "      \n",
            "      Earth's environments harbor complex consortia of microbes that affect processes ranging from host health to biogeochemical cycles. Understanding their evolution and function is limited by an inability to isolate genomes in a high-throughput manner. Here, we present a workflow for bacterial whole-genome sequencing using open-source labware and the OpenTrons robotics platform, reducing costs to approximately $10 per genome. We assess genomic diversity within 45 gut bacterial species from wild-living chimpanzees and bonobos. We quantify intraspecific genomic diversity and reveal divergence of homologous plasmids between hosts. This enables population genetic analyses of bacterial strains not currently possible with metagenomic data alone.\n",
            "    \n",
            "\n",
            "1  \n",
            "\n",
            "      \n",
            "      Common single-nucleotide polymorphisms (SNPs) are predicted to collectively explain 40-50% of phenotypic variation in human height, but identifying the specific variants and associated regions requires huge sample sizes1. Here, using data from a genome-wide association study of 5.4 million individuals of diverse ancestries, we show that 12,111 independent SNPs that are significantly associated with height account for nearly all of the common SNP-based heritability. These SNPs are clustered within 7,209 non-overlapping genomic segments with a mean size of around 90 kb, covering about 21% of the genome. The density of independent associations varies across the genome and the regions of increased density are enriched for biologically relevant genes. In out-of-sample estimation and prediction, the 12,111 SNPs (or all SNPs in the HapMap 3 panel2) account for 40% (45%) of phenotypic variance in populations of European ancestry but only around 10-20% (14-24%) in populations of other ancestries. Effect sizes, associated regions and gene prioritization are similar across ancestries, indicating that reduced prediction accuracy is likely to be explained by linkage disequilibrium and differences in allele frequency within associated regions. Finally, we show that the relevant biological pathways are detectable with smaller sample sizes than are needed to implicate causal genes and variants. Overall, this study provides a comprehensive map of specific genomic regions that contain the vast majority of common height-associated variants. Although this map is saturated for populations of European ancestry, further research is needed to achieve equivalent saturation in other ancestries.\n",
            "    \n",
            "\n",
            "2  \n",
            "\n",
            "\n",
            "          Objective:\n",
            "        \n",
            "      \n",
            "      To conduct a quality improvement evaluation of the Empower Veterans Program (EVP), an interdisciplinary pain rehabilitation/functional restoration program option for functional restoration for high-impact chronic pain, offered in a large metro-area Veterans Health Administration (VHA) system.\n",
            "    \n",
            "\n",
            "\n",
            "          Data sources:\n",
            "        \n",
            "      \n",
            "      VHA Corporate Data Warehouse electronic medical record data for patients treated by EVP between 2015 and 2019.\n",
            "    \n",
            "\n",
            "\n",
            "          Evaluation design:\n",
            "        \n",
            "      \n",
            "      This retrospective design first compared EVP patients considered engaged or not engaged in completing treatment in terms of demographic characteristics and post-treatment changes in clinical measures related to opioid use and mental health. We then compared mortality risk between matched groups of treated and untreated patients with chronic pain and concurrent opioid prescriptions using propensity score matching and Cox proportional hazards methods. \"Treated\" in the matched groups was defined as any level of EVP participation (i.e., both engaged and not engaged).\n",
            "    \n",
            "\n",
            "\n",
            "          Data collection/extraction methods:\n",
            "        \n",
            "      \n",
            "      We first identified 1053 EVP patients with 1 year of pre-and post-treatment follow-time and determined their engagement level. From those with chronic pain and prescription opioids, we matched 237 EVP patients to 375 untreated patients.\n",
            "    \n",
            "\n",
            "\n",
            "          Principal findings:\n",
            "        \n",
            "      \n",
            "      Engaged patients (57.4% of treated patients), were somewhat older than the non-engaged (mean age 57.1 vs. 53.7, Cohen's D = 0.30), and achieved lower mean PHQ9 depression scores in the post-treatment year (9.2 vs. 10.6, Cohen's D = 0.20). Participation in EVP was associated with a 65% lower mortality risk among Veterans with chronic pain and opioid use when compared to the untreated patients: (HR: 0.35, 95% CI: 0.17, 0.75).\n",
            "    \n",
            "\n",
            "\n",
            "          Conclusions:\n",
            "        \n",
            "      \n",
            "      EVP was associated with a large reduction in mortality risk for Veterans with both chronic pain and opioid use. This result could inform the decision process in a VA station or region when considering providing or expanding access to an interdisciplinary rehabilitation/functional restoration program for chronic pain.\n",
            "    \n",
            "\n",
            "3  \n",
            "\n",
            "      \n",
            "      Humans and Acanthamoeba polyphaga mimivirus share numerous homologous genes, including collagens and collagen-modifying enzymes. To explore this homology, we performed a genome-wide comparison between human and mimivirus using DELTA-BLAST (Domain Enhanced Lookup Time Accelerated BLAST) and identified 52 new putative mimiviral proteins that are homologous with human proteins. To gain functional insights into mimiviral proteins, their human protein homologs were organized into Gene Ontology (GO) and REACTOME pathways to build a functional network. Collagen and collagen-modifying enzymes form the largest subnetwork with most nodes. Further analysis of this subnetwork identified a putative collagen glycosyltransferase R699. Protein expression test suggested that R699 is highly expressed in Escherichia coli, unlike the human collagen-modifying enzymes. Enzymatic activity assay and mass spectrometric analyses showed that R699 catalyzes the glucosylation of galactosylhydroxylysine to glucosylgalactosylhydroxylysine on collagen using uridine diphosphate glucose (UDP-glucose) but no other UDP-sugars as a sugar donor, suggesting R699 is a mimiviral collagen galactosylhydroxylysyl glucosyltransferase (GGT). To facilitate further analysis of human and mimiviral homologous proteins, we presented an interactive and searchable genome-wide comparison website for quickly browsing human and Acanthamoeba polyphaga mimivirus homologs, which is available at RRID Resource ID: SCR_022140 or https://guolab.shinyapps.io/app-mimivirus-publication/ .\n",
            "    \n",
            "\n",
            "4 \n",
            "5  \n",
            "\n",
            "      \n",
            "      Adhesion G protein-coupled receptors (AGPCRs) are a family of 33 receptors in humans exhibiting a conserved general structure but diverse expression patterns and physiological functions. The large NH2 termini characteristic of AGPCRs confer unique properties to each receptor and possess a variety of distinct domains that can bind to a diverse array of extracellular proteins and components of the extracellular matrix. The traditional view of AGPCRs, as implied by their name, is that their core function is the mediation of adhesion. In recent years, though, many surprising advances have been made regarding AGPCR signaling mechanisms, activation by mechanosensory forces, and stimulation by small-molecule ligands such as steroid hormones and bioactive lipids. Thus, a new view of AGPCRs has begun to emerge in which these receptors are seen as massive signaling platforms that are crucial for the integration of adhesive, mechanosensory, and chemical stimuli. This review article describes the recent advances that have led to this new understanding of AGPCR function and also discusses new insights into the physiological actions of these receptors as well as their roles in human disease.\n",
            "    \n",
            "\n",
            "6 \n",
            "7  \n",
            "\n",
            "      \n",
            "      Previous genome-wide association studies (GWASs) of stroke - the second leading cause of death worldwide - were conducted predominantly in populations of European ancestry1,2. Here, in cross-ancestry GWAS meta-analyses of 110,182 patients who have had a stroke (five ancestries, 33% non-European) and 1,503,898 control individuals, we identify association signals for stroke and its subtypes at 89 (61 new) independent loci: 60 in primary inverse-variance-weighted analyses and 29 in secondary meta-regression and multitrait analyses. On the basis of internal cross-ancestry validation and an independent follow-up in 89,084 additional cases of stroke (30% non-European) and 1,013,843 control individuals, 87% of the primary stroke risk loci and 60% of the secondary stroke risk loci were replicated (P < 0.05). Effect sizes were highly correlated across ancestries. Cross-ancestry fine-mapping, in silico mutagenesis analysis3, and transcriptome-wide and proteome-wide association analyses revealed putative causal genes (such as SH3PXD2A and FURIN) and variants (such as at GRK5 and NOS3). Using a three-pronged approach4, we provide genetic evidence for putative drug effects, highlighting F11, KLKB1, PROC, GP1BA, LAMC2 and VCAM1 as possible targets, with drugs already under investigation for stroke for F11 and PROC. A polygenic score integrating cross-ancestry and ancestry-specific stroke GWASs with vascular-risk factor GWASs (integrative polygenic scores) strongly predicted ischaemic stroke in populations of European, East Asian and African ancestry5. Stroke genetic risk scores were predictive of ischaemic stroke independent of clinical risk factors in 52,600 clinical-trial participants with cardiometabolic disease. Our results provide insights to inform biology, reveal potential drug targets and derive genetic risk prediction tools across ancestries.\n",
            "    \n",
            "\n",
            "8  \n",
            "\n",
            "      \n",
            "      Many states' scope of practice laws limits the ability of nurse practitioners to deliver care by requiring physician supervision of their practices and prescribing activities. A robust literature has evolved around examining the role of these scope of practice laws in various contexts, including labor market outcomes, health care access, health care prices, and the delivery of care for specific diseases. Unfortunately, these studies use different, and sometimes conflicting, measures of scope of practice laws, limiting their comparability and overall usefulness to policymakers and future researchers. We address this salient problem by providing a recommended coding of nurse practitioner scope of practice laws over a 24-year period based on actual statutory and regulatory language. Our classification of scope of practice laws solves an important problem within this growing literature and provides a solid legal foundation for researchers as they continue to investigate the effects of these laws.\n",
            "    \n",
            "\n",
            "9  \n",
            "\n",
            "      \n",
            "      Pathogens trigger activation of sensors of the innate immune system that initiate molecular signaling enabling appropriate host defense programs. Although recognition of pathogen-specific moieties or PAMPs by specialized receptors of the immune system is well defined for a great number of pathogens, the mechanisms of sensing of pathogen-induced functional perturbations to the host cell remain poorly understood. Here we show that the disruption of endosomal compartments in macrophages by a bacterium or fully synthetic nanoparticles activates stress-response p38MAPK kinase, which triggers execution of cell death of a necrotic type. p38MAPK-mediated necrosis occurs in cells with a compound homozygous deletion of pyroptosis-inducing caspases-1 and -11, apoptotic caspase-8, and necroptosis-inducing receptor-interacting protein kinase-3 (RIPK3), indicating that all of these principal cell death mediators are dispensable for p38MAPK-induced necrosis in response to endosome rupture. p38MAPK-mediated necrosis is suppressed by the receptor-interacting protein kinase 1, RIPK1, and degradation of RIPK1 sensitizes macrophages to necrotic death. Since pathogen-induced cell death of necrotic types is implicated in host defense against infection, our results indicate that functional perturbations in host cells are sensed as a component of the innate immune system.\n",
            "    \n",
            "\n",
            "10  \n",
            "\n",
            "      \n",
            "      Expansion and differentiation of antigen-experienced PD-1+TCF-1+ stem-like CD8+ T cells into effector cells is critical for the success of immunotherapies based on PD-1 blockade1-4. Hashimoto et al. have shown that, in chronic infections, administration of the cytokine interleukin (IL)-2 triggers an alternative differentiation path of stem-like T cells towards a distinct population of 'better effector' CD8+ T cells similar to those generated in an acute infection5. IL-2 binding to the IL-2 receptor α-chain (CD25) was essential in triggering this alternative differentiation path and expanding better effectors with distinct transcriptional and epigenetic profiles. However, constitutive expression of CD25 on regulatory T cells and some endothelial cells also contributes to unwanted systemic effects from IL-2 therapy. Therefore, engineered IL-2 receptor β- and γ-chain (IL-2Rβγ)-biased agonists are currently being developed6-10. Here we show that IL-2Rβγ-biased agonists are unable to preferentially expand better effector T cells in cancer models and describe PD1-IL2v, a new immunocytokine that overcomes the need for CD25 binding by docking in cis to PD-1. Cis binding of PD1-IL2v to PD-1 and IL-2Rβγ on the same cell recovers the ability to differentiate stem-like CD8+ T cells into better effectors in the absence of CD25 binding in both chronic infection and cancer models and provides superior efficacy. By contrast, PD-1- or PD-L1-blocking antibodies alone, or their combination with clinically relevant doses of non-PD-1-targeted IL2v, cannot expand this unique subset of better effector T cells and instead lead to the accumulation of terminally differentiated, exhausted T cells. These findings provide the basis for the development of a new generation of PD-1 cis-targeted IL-2R agonists with enhanced therapeutic potential for the treatment of cancer and chronic infections.\n",
            "    \n",
            "\n",
            "11  \n",
            "\n",
            "      \n",
            "      Combination therapy with PD-1 blockade and IL-2 is highly effective during chronic lymphocytic choriomeningitis virus infection1. Here we examine the underlying basis for this synergy. We show that PD-1 + IL-2 combination therapy, in contrast to PD-1 monotherapy, substantially changes the differentiation program of the PD-1+TCF1+ stem-like CD8+ T cells and results in the generation of transcriptionally and epigenetically distinct effector CD8+ T cells that resemble highly functional effector CD8+ T cells seen after an acute viral infection. The generation of these qualitatively superior CD8+ T cells that mediate viral control underlies the synergy between PD-1 and IL-2. Our results show that the PD-1+TCF1+ stem-like CD8+ T cells, also referred to as precursors of exhausted CD8+ T cells, are not fate-locked into the exhaustion program and their differentiation trajectory can be changed by IL-2 signals. These virus-specific effector CD8+ T cells emerging from the stem-like CD8+ T cells after combination therapy expressed increased levels of the high-affinity IL-2 trimeric (CD25-CD122-CD132) receptor. This was not seen after PD-1 blockade alone. Finally, we show that CD25 engagement with IL-2 has an important role in the observed synergy between IL-2 cytokine and PD-1 blockade. Either blocking CD25 with an antibody or using a mutated version of IL-2 that does not bind to CD25 but still binds to CD122 and CD132 almost completely abrogated the synergistic effects observed after PD-1 + IL-2 combination therapy. There is considerable interest in PD-1 + IL-2 combination therapy for patients with cancer2,3, and our fundamental studies defining the underlying mechanisms of how IL-2 synergizes with PD-1 blockade should inform these human translational studies.\n",
            "    \n",
            "\n",
            "12  \n",
            "\n",
            "\n",
            "          Background:\n",
            "        \n",
            "      \n",
            "      The intrathecally administered antisense oligonucleotide tofersen reduces synthesis of the superoxide dismutase 1 (SOD1) protein and is being studied in patients with amyotrophic lateral sclerosis (ALS) associated with mutations in SOD1 (SOD1 ALS).\n",
            "    \n",
            "\n",
            "\n",
            "          Methods:\n",
            "        \n",
            "      \n",
            "      In this phase 3 trial, we randomly assigned adults with SOD1 ALS in a 2:1 ratio to receive eight doses of tofersen (100 mg) or placebo over a period of 24 weeks. The primary end point was the change from baseline to week 28 in the total score on the ALS Functional Rating Scale-Revised (ALSFRS-R; range, 0 to 48, with higher scores indicating better function) among participants predicted to have faster-progressing disease. Secondary end points included changes in the total concentration of SOD1 protein in cerebrospinal fluid (CSF), in the concentration of neurofilament light chains in plasma, in slow vital capacity, and in handheld dynamometry in 16 muscles. A combined analysis of the randomized component of the trial and its open-label extension at 52 weeks compared the results in participants who started tofersen at trial entry (early-start cohort) with those in participants who switched from placebo to the drug at week 28 (delayed-start cohort).\n",
            "    \n",
            "\n",
            "\n",
            "          Results:\n",
            "        \n",
            "      \n",
            "      A total of 72 participants received tofersen (39 predicted to have faster progression), and 36 received placebo (21 predicted to have faster progression). Tofersen led to greater reductions in concentrations of SOD1 in CSF and of neurofilament light chains in plasma than placebo. In the faster-progression subgroup (primary analysis), the change to week 28 in the ALSFRS-R score was -6.98 with tofersen and -8.14 with placebo (difference, 1.2 points; 95% confidence interval [CI], -3.2 to 5.5; P = 0.97). Results for secondary clinical end points did not differ significantly between the two groups. A total of 95 participants (88%) entered the open-label extension. At 52 weeks, the change in the ALSFRS-R score was -6.0 in the early-start cohort and -9.5 in the delayed-start cohort (difference, 3.5 points; 95% CI, 0.4 to 6.7); non-multiplicity-adjusted differences favoring early-start tofersen were seen for other end points. Lumbar puncture-related adverse events were common. Neurologic serious adverse events occurred in 7% of tofersen recipients.\n",
            "    \n",
            "\n",
            "\n",
            "          Conclusions:\n",
            "        \n",
            "      \n",
            "      In persons with SOD1 ALS, tofersen reduced concentrations of SOD1 in CSF and of neurofilament light chains in plasma over 28 weeks but did not improve clinical end points and was associated with adverse events. The potential effects of earlier as compared with delayed initiation of tofersen are being further evaluated in the extension phase. (Funded by Biogen; VALOR and OLE ClinicalTrials.gov numbers, NCT02623699 and NCT03070119; EudraCT numbers, 2015-004098-33 and 2016-003225-41.).\n",
            "    \n",
            "\n",
            "13  \n",
            "\n",
            "      \n",
            "      Medulloblastoma (MB) comprises a group of heterogeneous paediatric embryonal neoplasms of the hindbrain with strong links to early development of the hindbrain1-4. Mutations that activate Sonic hedgehog signalling lead to Sonic hedgehog MB in the upper rhombic lip (RL) granule cell lineage5-8. By contrast, mutations that activate WNT signalling lead to WNT MB in the lower RL9,10. However, little is known about the more commonly occurring group 4 (G4) MB, which is thought to arise in the unipolar brush cell lineage3,4. Here we demonstrate that somatic mutations that cause G4 MB converge on the core binding factor alpha (CBFA) complex and mutually exclusive alterations that affect CBFA2T2, CBFA2T3, PRDM6, UTX and OTX2. CBFA2T2 is expressed early in the progenitor cells of the cerebellar RL subventricular zone in Homo sapiens, and G4 MB transcriptionally resembles these progenitors but are stalled in developmental time. Knockdown of OTX2 in model systems relieves this differentiation blockade, which allows MB cells to spontaneously proceed along normal developmental differentiation trajectories. The specific nature of the split human RL, which is destined to generate most of the neurons in the human brain, and its high level of susceptible EOMES+KI67+ unipolar brush cell progenitor cells probably predisposes our species to the development of G4 MB.\n",
            "    \n",
            "\n",
            "14  \n",
            "\n",
            "      \n",
            "      Germinal centres are the engines of antibody evolution. Here, using human immunodeficiency virus (HIV) Env protein immunogen priming in rhesus monkeys followed by a long period without further immunization, we demonstrate germinal centre B (BGC) cells that last for at least 6 months. A 186-fold increase in BGC cells was present by week 10 compared with conventional immunization. Single-cell transcriptional profiling showed that both light- and dark-zone germinal centre states were sustained. Antibody somatic hypermutation of BGC cells continued to accumulate throughout the 29-week priming period, with evidence of selective pressure. Env-binding BGC cells were still 49-fold above baseline at 29 weeks, which suggests that they could remain active for even longer periods of time. High titres of HIV-neutralizing antibodies were generated after a single booster immunization. Fully glycosylated HIV trimer protein is a complex antigen, posing considerable immunodominance challenges for B cells1,2. Memory B cells generated under these long priming conditions had higher levels of antibody somatic hypermutation, and both memory B cells and antibodies were more likely to recognize non-immunodominant epitopes. Numerous BGC cell lineage phylogenies spanning more than the 6-month germinal centre period were identified, demonstrating continuous germinal centre activity and selection for at least 191 days with no further antigen exposure. A long-prime, slow-delivery (12 days) immunization approach holds promise for difficult vaccine targets and suggests that patience can have great value for tuning of germinal centres to maximize antibody responses.\n",
            "    \n",
            "\n",
            "15  \n",
            "\n",
            "      \n",
            "      Seven viruses cause at least 15% of the total cancer burden. Viral cancers have been described as the \"low-hanging fruit\" that can be potentially prevented or treated by new vaccines that would alter the course of global human cancer. Kaposi sarcoma herpesvirus (KSHV or HHV8) is the sole cause of Kaposi sarcoma, which primarily afflicts resource-poor and socially marginalized populations. This review summarizes a recent NIH-sponsored workshop's findings on the epidemiology and biology of KSHV as an overlooked but potentially vaccine-preventable infection. The unique epidemiology of this virus provides opportunities to prevent its cancers if an effective, inexpensive, and well-tolerated vaccine can be developed and delivered.\n",
            "    \n",
            "\n",
            "16  \n",
            "\n",
            "\n",
            "          Purpose:\n",
            "        \n",
            "      \n",
            "      To update the ASCO Biomarkers to Guide Systemic Therapy for Metastatic Breast Cancer (MBC) guideline.\n",
            "    \n",
            "\n",
            "\n",
            "          Methods:\n",
            "        \n",
            "      \n",
            "      An Expert Panel conducted a systematic review to identify randomized clinical trials and prospective-retrospective studies from January 2015 to January 2022.\n",
            "    \n",
            "\n",
            "\n",
            "          Results:\n",
            "        \n",
            "      \n",
            "      The search identified 19 studies informing the evidence base.\n",
            "    \n",
            "\n",
            "\n",
            "          Recommendations:\n",
            "        \n",
            "      \n",
            "      Candidates for a regimen with a phosphatidylinositol 3-kinase inhibitor and hormonal therapy should undergo testing for PIK3CA mutations using next-generation sequencing of tumor tissue or circulating tumor DNA (ctDNA) in plasma to determine eligibility for alpelisib plus fulvestrant. If no mutation is found in ctDNA, testing in tumor tissue, if available, should be used. Patients who are candidates for poly (ADP-ribose) polymerase (PARP) inhibitor therapy should undergo testing for germline BRCA1 and BRCA2 pathogenic or likely pathogenic mutations to determine eligibility for a PARP inhibitor. There is insufficient evidence for or against testing for a germline PALB2 pathogenic variant to determine eligibility for PARP inhibitor therapy in the metastatic setting. Candidates for immune checkpoint inhibitor therapy should undergo testing for expression of programmed cell death ligand-1 in the tumor and immune cells to determine eligibility for treatment with pembrolizumab plus chemotherapy. Candidates for an immune checkpoint inhibitor should also undergo testing for deficient mismatch repair/microsatellite instability-high to determine eligibility for dostarlimab-gxly or pembrolizumab, as well as testing for tumor mutational burden. Clinicians may test for NTRK fusions to determine eligibility for TRK inhibitors. There are insufficient data to recommend routine testing of tumors for ESR1 mutations, for homologous recombination deficiency, or for TROP2 expression to guide MBC therapy selection. There are insufficient data to recommend routine use of ctDNA or circulating tumor cells to monitor response to therapy among patients with MBC.Additional information can be found at www.asco.org/breast-cancer-guidelines.\n",
            "    \n",
            "\n",
            "17  \n",
            "\n",
            "Clinical trials frequently include multiple end points that mature at different times. The initial report, typically based on the primary end point, may be published when key planned coprimary or secondary analyses are not yet available. Clinical Trial Updates provide an opportunity to disseminate additional results from studies, published in JCO or elsewhere, for which the primary end point has already been reported.The purpose of this update was to determine differences in patient-reported chronic toxicity and disease outcomes with intensity-modulated radiation therapy (IMRT) compared with conventional pelvic radiation. Patients with cervical and endometrial cancers who received postoperative pelvic radiation were randomly assigned to conventional radiation therapy (CRT) or IMRT. Toxicity and quality of life were assessed using Patient-Reported Outcomes version of the Common Terminology Criteria for Adverse Events, Expanded Prostate Cancer Index Composite (EPIC) bowel and urinary domains, and Functional Assessment of Cancer Therapy-General. Between 2012 and 2015, 279 eligible patients were enrolled to the study with a median follow-up of 37.8 months. There were no differences in overall survival (P = .53), disease-free survival (P = .21), or locoregional failure (P = .81). One year after RT, patients in the CRT arm experienced more high-level diarrhea frequency (5.8% IMRT v 15.1% CRT, P = .042) and a greater number had to take antidiarrheal medication two or more times a day (1.2% IMRT v 8.6% CRT, P = .036). At 3 years, women in the CRT arm reported a decline in urinary function, whereas the IMRT arm continued to improve (mean change in EPIC urinary score = 0.5, standard deviation = 13.0, IMRT v -6.0, standard deviation = 14.3, CRT, P = .005). In conclusion, IMRT reduces patient-reported chronic GI and urinary toxicity with no difference in treatment efficacy at 3 years.\n",
            "    \n",
            "\n",
            "18  \n",
            "\n",
            "      \n",
            "      Air pollution is a major environmental and public health challenge in China and the Chinese government has implemented a series of strict air quality policies. However, particulate nitrate (NO3 -) concentration remains high or even increases at monitoring sites despite the total PM2.5 concentration has decreased. Unfortunately, it has been difficult to estimate NO3 - concentration across China due to the lack of a PM2.5 speciation monitoring network. Here, we use a machine learning model incorporating ground measurements and satellite data to characterize the spatiotemporal patterns of NO3 -, thereby understanding the disease burden associated with long-term NO3 - exposure in China. Our results show that existing air pollution control policies are effective, but increased NO3 - of traffic emissions offset reduced NO3 - of industrial emissions. In 2018, the national mean mortality burden attributable to NO3 - was as high as 0.68 million, indicating that targeted regulations are needed to control NO3 - pollution in China.\n",
            "    \n",
            "\n",
            "19  \n",
            "\n",
            "      \n",
            "      Idiopathic scoliosis (IS) refers to abnormal spinal curvatures that occur in the absence of vertebral or neuromuscular defects. IS accounts for 80% of human spinal deformity, afflicts ∼3% of children worldwide, yet pathogenic mechanisms are poorly understood. A key role for cerebrospinal fluid (CSF) homeostasis in zebrafish spine development has been identified. Specifically, defects in cilia motility of brain ependymal cells (EC), CSF flow, and/or Reissner fiber (RF) assembly are observed to induce neuroinflammation, oxidative stress, abnormal CSF-contacting neuron activity, and urotensin peptide expression, all associating with scoliosis. However, the functional relevance of these observations to IS remains unclear. Here we characterize zebrafish katnb1 mutants as a new IS model. We define essential roles for Katnb1 in motile ciliated lineages, uncouple EC cilia and RF formation defects from spinal curvature, and identify abnormal CSF flow and cell stress responses as shared pathogenic signatures associated with scoliosis across diverse zebrafish models.\n",
            "    \n",
            "\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:11: SettingWithCopyWarning: \n",
            "A value is trying to be set on a copy of a slice from a DataFrame.\n",
            "Try using .loc[row_indexer,col_indexer] = value instead\n",
            "\n",
            "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
            "  # This is added back by InteractiveShellApp.init_path()\n"
          ]
        }
      ],
      "source": [
        "d = driversetup()\n",
        "abstracts = []\n",
        "for i in range(len(sample)):\n",
        "  s = pagesource(sample.loc[i]['URL'], d)\n",
        "  abstract = list(s.findAll('div', class_=\"abstract-content selected\"))\n",
        "  a = ''\n",
        "  for j in range(len(abstract)):\n",
        "    a = a + ' ' + abstract[j].text\n",
        "  abstracts.append(a)\n",
        "  print(i, a)\n",
        "sample[\"PubMed_Abstracts\"] = abstracts"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "sample"
      ],
      "metadata": {
        "id": "JJg0DuUpOLEk",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        },
        "outputId": "9ba08668-b08d-497c-fa74-b9f00327cfb4"
      },
      "id": "JJg0DuUpOLEk",
      "execution_count": 15,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "                                        Article Title  \\\n",
              "0   A low-cost genomics workflow enables isolate s...   \n",
              "1   A saturated map of common genetic variants ass...   \n",
              "2   Evaluation of a non-pharmacological interdisci...   \n",
              "3   Comparative genomic and biochemical analyses i...   \n",
              "4   Africa must participate in finding a gene ther...   \n",
              "5   ADHESION G PROTEIN-COUPLED RECEPTORS: STRUCTUR...   \n",
              "6   Authorship equity guidelines in global health ...   \n",
              "7   Stroke genetics informs drug discovery and ris...   \n",
              "8   Toward a Uniform Classification of Nurse Pract...   \n",
              "9   p38MAPK guards the integrity of endosomal comp...   \n",
              "10  PD-1-cis IL-2R agonism yields better effectors...   \n",
              "11  PD-1 combination therapy with IL-2 modifies CD...   \n",
              "12  Trial of Antisense Oligonucleotide Tofersen fo...   \n",
              "13  Failure of human rhombic lip differentiation u...   \n",
              "14  Long-primed germinal centres with enduring aff...   \n",
              "15  KSHV (HHV8) vaccine: promises and potential pi...   \n",
              "16  Biomarkers for Systemic Therapy in Metastatic ...   \n",
              "17  Intensity-Modulated Radiation Therapy Reduces ...   \n",
              "18  Assessment of long-term particulate nitrate ai...   \n",
              "19  Resolving primary pathomechanisms driving idio...   \n",
              "\n",
              "                                             Abstract   Pubmed Id  \\\n",
              "0   Earth's environments harbor complex consortia ...  36224660.0   \n",
              "1   Common single-nucleotide polymorphisms (SNPs) ...  36224396.0   \n",
              "2   Objective To conduct a quality improvement eva...  36064854.0   \n",
              "3   Humans and Acanthamoeba polyphaga mimivirus sh...  36207453.0   \n",
              "4                                                 NaN  36203001.0   \n",
              "5   Adhesion G protein-coupled receptors (AGPCRs) ...  35468004.0   \n",
              "6                                                 NaN  36220309.0   \n",
              "7   Previous genome-wide association studies (GWAS...  36180795.0   \n",
              "8   Many states' scope of practice laws limits the...  36172783.0   \n",
              "9   Pathogens trigger activation of sensors of the...  36175595.0   \n",
              "10  Expansion and differentiation of antigen-exper...  36171284.0   \n",
              "11  Combination therapy with PD-1 blockade and IL-...  36171288.0   \n",
              "12  BackgroundThe intrathecally administered antis...  36129998.0   \n",
              "13  Medulloblastoma (MB) comprises a group of hete...  36131014.0   \n",
              "14  Germinal centres are the engines of antibody e...  36131022.0   \n",
              "15  Seven viruses cause at least 15% of the total ...  36127367.0   \n",
              "16  PUPPOSE To update the ASCO Biomarkers to Guide...  35759724.0   \n",
              "17  Clinical trials frequently include multiple en...  35960897.0   \n",
              "18  Air pollution is a major environmental and pub...  36039292.0   \n",
              "19  Idiopathic scoliosis (IS) refers to abnormal s...  36105588.0   \n",
              "\n",
              "                                          URL  \\\n",
              "0   https://pubmed.ncbi.nlm.nih.gov/36224660/   \n",
              "1   https://pubmed.ncbi.nlm.nih.gov/36224396/   \n",
              "2   https://pubmed.ncbi.nlm.nih.gov/36064854/   \n",
              "3   https://pubmed.ncbi.nlm.nih.gov/36207453/   \n",
              "4   https://pubmed.ncbi.nlm.nih.gov/36203001/   \n",
              "5   https://pubmed.ncbi.nlm.nih.gov/35468004/   \n",
              "6   https://pubmed.ncbi.nlm.nih.gov/36220309/   \n",
              "7   https://pubmed.ncbi.nlm.nih.gov/36180795/   \n",
              "8   https://pubmed.ncbi.nlm.nih.gov/36172783/   \n",
              "9   https://pubmed.ncbi.nlm.nih.gov/36175595/   \n",
              "10  https://pubmed.ncbi.nlm.nih.gov/36171284/   \n",
              "11  https://pubmed.ncbi.nlm.nih.gov/36171288/   \n",
              "12  https://pubmed.ncbi.nlm.nih.gov/36129998/   \n",
              "13  https://pubmed.ncbi.nlm.nih.gov/36131014/   \n",
              "14  https://pubmed.ncbi.nlm.nih.gov/36131022/   \n",
              "15  https://pubmed.ncbi.nlm.nih.gov/36127367/   \n",
              "16  https://pubmed.ncbi.nlm.nih.gov/35759724/   \n",
              "17  https://pubmed.ncbi.nlm.nih.gov/35960897/   \n",
              "18  https://pubmed.ncbi.nlm.nih.gov/36039292/   \n",
              "19  https://pubmed.ncbi.nlm.nih.gov/36105588/   \n",
              "\n",
              "                                     PubMed_Abstracts  \n",
              "0    \\n\\n      \\n      Earth's environments harbor...  \n",
              "1    \\n\\n      \\n      Common single-nucleotide po...  \n",
              "2    \\n\\n\\n          Objective:\\n        \\n      \\...  \n",
              "3    \\n\\n      \\n      Humans and Acanthamoeba pol...  \n",
              "4                                                      \n",
              "5    \\n\\n      \\n      Adhesion G protein-coupled ...  \n",
              "6                                                      \n",
              "7    \\n\\n      \\n      Previous genome-wide associ...  \n",
              "8    \\n\\n      \\n      Many states' scope of pract...  \n",
              "9    \\n\\n      \\n      Pathogens trigger activatio...  \n",
              "10   \\n\\n      \\n      Expansion and differentiati...  \n",
              "11   \\n\\n      \\n      Combination therapy with PD...  \n",
              "12   \\n\\n\\n          Background:\\n        \\n      ...  \n",
              "13   \\n\\n      \\n      Medulloblastoma (MB) compri...  \n",
              "14   \\n\\n      \\n      Germinal centres are the en...  \n",
              "15   \\n\\n      \\n      Seven viruses cause at leas...  \n",
              "16   \\n\\n\\n          Purpose:\\n        \\n      \\n ...  \n",
              "17   \\n\\nClinical trials frequently include multip...  \n",
              "18   \\n\\n      \\n      Air pollution is a major en...  \n",
              "19   \\n\\n      \\n      Idiopathic scoliosis (IS) r...  "
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-11d30bd6-188d-4187-b1de-662fbbc97c30\">\n",
              "    <div class=\"colab-df-container\">\n",
              "      <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>Article Title</th>\n",
              "      <th>Abstract</th>\n",
              "      <th>Pubmed Id</th>\n",
              "      <th>URL</th>\n",
              "      <th>PubMed_Abstracts</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>A low-cost genomics workflow enables isolate s...</td>\n",
              "      <td>Earth's environments harbor complex consortia ...</td>\n",
              "      <td>36224660.0</td>\n",
              "      <td>https://pubmed.ncbi.nlm.nih.gov/36224660/</td>\n",
              "      <td>\\n\\n      \\n      Earth's environments harbor...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>A saturated map of common genetic variants ass...</td>\n",
              "      <td>Common single-nucleotide polymorphisms (SNPs) ...</td>\n",
              "      <td>36224396.0</td>\n",
              "      <td>https://pubmed.ncbi.nlm.nih.gov/36224396/</td>\n",
              "      <td>\\n\\n      \\n      Common single-nucleotide po...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>Evaluation of a non-pharmacological interdisci...</td>\n",
              "      <td>Objective To conduct a quality improvement eva...</td>\n",
              "      <td>36064854.0</td>\n",
              "      <td>https://pubmed.ncbi.nlm.nih.gov/36064854/</td>\n",
              "      <td>\\n\\n\\n          Objective:\\n        \\n      \\...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>Comparative genomic and biochemical analyses i...</td>\n",
              "      <td>Humans and Acanthamoeba polyphaga mimivirus sh...</td>\n",
              "      <td>36207453.0</td>\n",
              "      <td>https://pubmed.ncbi.nlm.nih.gov/36207453/</td>\n",
              "      <td>\\n\\n      \\n      Humans and Acanthamoeba pol...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>Africa must participate in finding a gene ther...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>36203001.0</td>\n",
              "      <td>https://pubmed.ncbi.nlm.nih.gov/36203001/</td>\n",
              "      <td></td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5</th>\n",
              "      <td>ADHESION G PROTEIN-COUPLED RECEPTORS: STRUCTUR...</td>\n",
              "      <td>Adhesion G protein-coupled receptors (AGPCRs) ...</td>\n",
              "      <td>35468004.0</td>\n",
              "      <td>https://pubmed.ncbi.nlm.nih.gov/35468004/</td>\n",
              "      <td>\\n\\n      \\n      Adhesion G protein-coupled ...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>6</th>\n",
              "      <td>Authorship equity guidelines in global health ...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>36220309.0</td>\n",
              "      <td>https://pubmed.ncbi.nlm.nih.gov/36220309/</td>\n",
              "      <td></td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>7</th>\n",
              "      <td>Stroke genetics informs drug discovery and ris...</td>\n",
              "      <td>Previous genome-wide association studies (GWAS...</td>\n",
              "      <td>36180795.0</td>\n",
              "      <td>https://pubmed.ncbi.nlm.nih.gov/36180795/</td>\n",
              "      <td>\\n\\n      \\n      Previous genome-wide associ...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>8</th>\n",
              "      <td>Toward a Uniform Classification of Nurse Pract...</td>\n",
              "      <td>Many states' scope of practice laws limits the...</td>\n",
              "      <td>36172783.0</td>\n",
              "      <td>https://pubmed.ncbi.nlm.nih.gov/36172783/</td>\n",
              "      <td>\\n\\n      \\n      Many states' scope of pract...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>9</th>\n",
              "      <td>p38MAPK guards the integrity of endosomal comp...</td>\n",
              "      <td>Pathogens trigger activation of sensors of the...</td>\n",
              "      <td>36175595.0</td>\n",
              "      <td>https://pubmed.ncbi.nlm.nih.gov/36175595/</td>\n",
              "      <td>\\n\\n      \\n      Pathogens trigger activatio...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>10</th>\n",
              "      <td>PD-1-cis IL-2R agonism yields better effectors...</td>\n",
              "      <td>Expansion and differentiation of antigen-exper...</td>\n",
              "      <td>36171284.0</td>\n",
              "      <td>https://pubmed.ncbi.nlm.nih.gov/36171284/</td>\n",
              "      <td>\\n\\n      \\n      Expansion and differentiati...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11</th>\n",
              "      <td>PD-1 combination therapy with IL-2 modifies CD...</td>\n",
              "      <td>Combination therapy with PD-1 blockade and IL-...</td>\n",
              "      <td>36171288.0</td>\n",
              "      <td>https://pubmed.ncbi.nlm.nih.gov/36171288/</td>\n",
              "      <td>\\n\\n      \\n      Combination therapy with PD...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>12</th>\n",
              "      <td>Trial of Antisense Oligonucleotide Tofersen fo...</td>\n",
              "      <td>BackgroundThe intrathecally administered antis...</td>\n",
              "      <td>36129998.0</td>\n",
              "      <td>https://pubmed.ncbi.nlm.nih.gov/36129998/</td>\n",
              "      <td>\\n\\n\\n          Background:\\n        \\n      ...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>13</th>\n",
              "      <td>Failure of human rhombic lip differentiation u...</td>\n",
              "      <td>Medulloblastoma (MB) comprises a group of hete...</td>\n",
              "      <td>36131014.0</td>\n",
              "      <td>https://pubmed.ncbi.nlm.nih.gov/36131014/</td>\n",
              "      <td>\\n\\n      \\n      Medulloblastoma (MB) compri...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>14</th>\n",
              "      <td>Long-primed germinal centres with enduring aff...</td>\n",
              "      <td>Germinal centres are the engines of antibody e...</td>\n",
              "      <td>36131022.0</td>\n",
              "      <td>https://pubmed.ncbi.nlm.nih.gov/36131022/</td>\n",
              "      <td>\\n\\n      \\n      Germinal centres are the en...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>15</th>\n",
              "      <td>KSHV (HHV8) vaccine: promises and potential pi...</td>\n",
              "      <td>Seven viruses cause at least 15% of the total ...</td>\n",
              "      <td>36127367.0</td>\n",
              "      <td>https://pubmed.ncbi.nlm.nih.gov/36127367/</td>\n",
              "      <td>\\n\\n      \\n      Seven viruses cause at leas...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>16</th>\n",
              "      <td>Biomarkers for Systemic Therapy in Metastatic ...</td>\n",
              "      <td>PUPPOSE To update the ASCO Biomarkers to Guide...</td>\n",
              "      <td>35759724.0</td>\n",
              "      <td>https://pubmed.ncbi.nlm.nih.gov/35759724/</td>\n",
              "      <td>\\n\\n\\n          Purpose:\\n        \\n      \\n ...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>17</th>\n",
              "      <td>Intensity-Modulated Radiation Therapy Reduces ...</td>\n",
              "      <td>Clinical trials frequently include multiple en...</td>\n",
              "      <td>35960897.0</td>\n",
              "      <td>https://pubmed.ncbi.nlm.nih.gov/35960897/</td>\n",
              "      <td>\\n\\nClinical trials frequently include multip...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>18</th>\n",
              "      <td>Assessment of long-term particulate nitrate ai...</td>\n",
              "      <td>Air pollution is a major environmental and pub...</td>\n",
              "      <td>36039292.0</td>\n",
              "      <td>https://pubmed.ncbi.nlm.nih.gov/36039292/</td>\n",
              "      <td>\\n\\n      \\n      Air pollution is a major en...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>19</th>\n",
              "      <td>Resolving primary pathomechanisms driving idio...</td>\n",
              "      <td>Idiopathic scoliosis (IS) refers to abnormal s...</td>\n",
              "      <td>36105588.0</td>\n",
              "      <td>https://pubmed.ncbi.nlm.nih.gov/36105588/</td>\n",
              "      <td>\\n\\n      \\n      Idiopathic scoliosis (IS) r...</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "      <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-11d30bd6-188d-4187-b1de-662fbbc97c30')\"\n",
              "              title=\"Convert this dataframe to an interactive table.\"\n",
              "              style=\"display:none;\">\n",
              "        \n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M0 0h24v24H0V0z\" fill=\"none\"/>\n",
              "    <path d=\"M18.56 5.44l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94zm-11 1L8.5 8.5l.94-2.06 2.06-.94-2.06-.94L8.5 2.5l-.94 2.06-2.06.94zm10 10l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94z\"/><path d=\"M17.41 7.96l-1.37-1.37c-.4-.4-.92-.59-1.43-.59-.52 0-1.04.2-1.43.59L10.3 9.45l-7.72 7.72c-.78.78-.78 2.05 0 2.83L4 21.41c.39.39.9.59 1.41.59.51 0 1.02-.2 1.41-.59l7.78-7.78 2.81-2.81c.8-.78.8-2.07 0-2.86zM5.41 20L4 18.59l7.72-7.72 1.47 1.35L5.41 20z\"/>\n",
              "  </svg>\n",
              "      </button>\n",
              "      \n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      flex-wrap:wrap;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "      <script>\n",
              "        const buttonEl =\n",
              "          document.querySelector('#df-11d30bd6-188d-4187-b1de-662fbbc97c30 button.colab-df-convert');\n",
              "        buttonEl.style.display =\n",
              "          google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "        async function convertToInteractive(key) {\n",
              "          const element = document.querySelector('#df-11d30bd6-188d-4187-b1de-662fbbc97c30');\n",
              "          const dataTable =\n",
              "            await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                     [key], {});\n",
              "          if (!dataTable) return;\n",
              "\n",
              "          const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "            '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "            + ' to learn more about interactive tables.';\n",
              "          element.innerHTML = '';\n",
              "          dataTable['output_type'] = 'display_data';\n",
              "          await google.colab.output.renderOutput(dataTable, element);\n",
              "          const docLink = document.createElement('div');\n",
              "          docLink.innerHTML = docLinkHtml;\n",
              "          element.appendChild(docLink);\n",
              "        }\n",
              "      </script>\n",
              "    </div>\n",
              "  </div>\n",
              "  "
            ]
          },
          "metadata": {},
          "execution_count": 15
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [],
      "metadata": {
        "id": "2AxjBX_8izMg"
      },
      "id": "2AxjBX_8izMg",
      "execution_count": null,
      "outputs": []
    }
  ],
  "metadata": {
    "kernelspec": {
      "display_name": "Python 3 (ipykernel)",
      "language": "python",
      "name": "python3"
    },
    "language_info": {
      "codemirror_mode": {
        "name": "ipython",
        "version": 3
      },
      "file_extension": ".py",
      "mimetype": "text/x-python",
      "name": "python",
      "nbconvert_exporter": "python",
      "pygments_lexer": "ipython3",
      "version": "3.9.7"
    },
    "colab": {
      "provenance": [],
      "include_colab_link": true
    }
  },
  "nbformat": 4,
  "nbformat_minor": 5
}